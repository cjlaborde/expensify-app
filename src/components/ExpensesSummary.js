import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'
import selectExpenseTotal from '../selectors/expenses-total'

// import getExpenseTotal from '../selectors/expenses-total';

const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount > 1 ? 'expenses' : 'expense'
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        
        </div>
    )
      
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpenseTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)