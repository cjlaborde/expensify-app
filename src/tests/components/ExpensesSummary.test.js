import React from 'react'
import { shallow } from 'enzyme'
import ExpensesSummary from '../../components/ExpensesSummary';
// import { Link } from 'react-router-dom'

test('Should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary />)
    expect(wrapper).toMatchSnapshot()
})

test('Should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={235543535}/>)
    expect(wrapper).toMatchSnapshot()
})

