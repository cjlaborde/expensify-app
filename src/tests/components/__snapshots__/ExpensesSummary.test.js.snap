// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render ExpensesSummary with 1 expense 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Should correctly render ExpensesSummary with 1 expense 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
